### ISSUE TITLE
`group-exports`: option for export at definition

### ISSUE THREAD

**Issue by golopot** [import-js/eslint-plugin-import#2406]
The option enforces exports statements to to happens at definition:

**Pass:**
```js
export default function foo() {}
export function foo(){}
export const foo = 1;
```

**Fail:**
```js
function foo() {}
export {
  foo,
}
```

The default option for `group-exports` will be called `always`, and the added option will be `never`.

**Comment by ljharb** [comment]
This sounds like it might have a ton of unintended consequences.

group-exports is generally about the position in the file, not the form used to export. There’s all sorts of reasons why you might not be *able* to use the declaration/inline form, and why the curly-braced form is the only way you can export something. I worry such an option would be harmful.

**Comment by golopot** [comment]
What are some examples where curly based exports are necessary? I always use inline exports at my work and have not met a problem.

I prefer the inline form because I can tell whether the function is exported right at the definition, instead of having to find out whether it is exported elsewhere.

As a side note the inline syntax is the only way to export things in Golang.

**Comment by golopot** [comment]
I thought of one case where I use curly braced exports: reexports.

**Comment by ljharb** [comment]
JS isn’t go, so im not sure that’s relevant; it’s a counterproductive goal to match coding styles across languages.

If the value being exported isn’t a declaration. Maybe it’s the return value of a function, or maybe it’s imported from somewhere else.
